tests/test_interpolate.py

```diff
@@ -0,0 +1,227 @@
+from jax.config import config
+
+config.update("jax_enable_x64", True)
+
+from importlib import resources
+
+import numpy as np
+from rdkit import Chem
+
+from timemachine.fe import atom_mapping, interpolate, single_topology_v3
+from timemachine.fe.utils import get_romol_conf
+from timemachine.ff import Forcefield
+
+
+def test_align_harmonic_bond():
+    """
+    Test that we can align idxs and parameters correctly for harmonic bonds.
+    We expect that decoupled terms have their force constants turned set to zero,
+    while maintaining the same equilibrium bond lengths.
+    """
+    a, b, c, d, e, f, g, h, i, j = np.random.rand(10)
```

Comment:
I don't think this really matters that much - this could be literally just `np.arange(10)` if that's more preferable

Reply:
This test is still deterministic in the output even if a random number is used in the input. 

Consider: you can have a function f(x)=0 have random inputs but the output will always be zero. This is a deterministic output.

Furthermore, you can have a function f(x)=x assert that output=input for any value of x (even random ones). The behavior is completely deterministic. This is what the current usage pattern is, ie we're just asserting that the broadcasting is done correctly on some arbitrary values. 

We require determinism tests for nonbonded tests and randomly generated inputs because there are two sources of non-determinism:

The test function f(x) may return different values if we do not use bitwise checks (for the same value of x). 

The test function f(x) may not agree with a reference function g(x) to a sufficient level of precision for different values of x.

Reply:
I agree that this seems minor, but just wanted to clarify my reasoning for the initial comment.

Deterministic inputs seem preferable to me, even in cases where we are confident that the outcome is regardless deterministic, because
- We don't need to rely on (potentially flawed, especially in more complex cases) reasoning that the result is indeed deterministic. With nondeterministic inputs, the test might seem suspicious to future readers expecting deterministic tests until they've verified this reasoning for themselves
- There doesn't seem to be a benefit to nondeterministic inputs. If we want to test that the outcome is satisfied for varying inputs, we can still do this deterministically by generating a fixed number of random variates, as @maxentile mentioned